pipelines/legionPipeline.groovy

```diff
@@ -427,71 +431,71 @@ def runRobotTestsAtGcp(tags="") {
     withCredentials([
     file(credentialsId: "${env.gcpCredential}", variable: 'gcpCredential')]) {
         withCredentials([
-        file(credentialsId: "${env.param_cluster_name}-testvars", variable: 'testcreds')]) {
-            withAWS(credentials: 'kops') {
-                wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
-                    docker.image("${env.param_docker_repo}/legion-pipeline-agent:${env.param_legion_version}").inside("-e HOME=/opt/legion -u root") {
-                        stage('Run Robot tests') {
-                            dir("${WORKSPACE}"){
-                                def tags_list = tags.toString().trim().split(',')
-                                def robot_tags = []
-                                def nose_tags = []
-
-                                for (item in tags_list) {
-                                    if (item.startsWith('-')) {
-                                        item = item.replace("-","")
-                                        robot_tags.add(" -e ${item}")
-                                        nose_tags.add(" -a !${item}")
+        file(credentialsId: "${env.hieraPrivatePKCSKey}", variable: 'PrivatePkcsKey')]) {
+            withCredentials([
+            file(credentialsId: "${env.hieraPublicPKCSKey}", variable: 'PublicPkcsKey')]) {
+                withAWS(credentials: 'kops') {
+                    wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
+                        docker.image("${env.param_docker_repo}/legion-pipeline-agent:${env.param_legion_version}").inside("-e HOME=/opt/legion -u root") {
+                            stage('Extract Hiera data') {
+                                extractHiera("json")
```

Comment:
IMHO, it's a bad practice when our (legion\airflow\mlflow and so on) non-infra repositories know about hiera converter and its dependencies. For example, you install ruby, rubygems, hiera and pyyaml in the legion pipeline docker image https://github.com/legion-platform/legion/pull/970/files#diff-02048042aaff3c36726a71b3aa9a1fccR21 .
What do you think about the following way for robot tests?
```groovy
docker.image("infrasture-image") {
  extractHiera("json")
}
docker.image("legion-pipeline-agent") {
  run_robot_test()
}
```